test/gui/shared/steps/steps.py

```diff
@@ -447,32 +533,40 @@ def collaboratorShouldBeListed(
 
 @When('the user waits for the files to sync')
 def step(context):
-    waitForRootFolderToSync(context)
+    waitForSyncToComplete(context)
 
 
 @When(r'the user waits for (file|folder) "([^"]*)" to be synced', regexp=True)
 def step(context, type, resource):
-    waitForFileOrFolderToSync(context, resource, type)
+    waitForSyncToComplete(context, resource, type)
 
 
 @When(r'the user waits for (file|folder) "([^"]*)" to have sync error', regexp=True)
 def step(context, type, resource):
     waitForFileOrFolderToHaveSyncError(context, resource, type)
 
 
+@When(
+    r'user "([^"]*)" waits for (file|folder) "([^"]*)" to have sync error', regexp=True
+)
+def step(context, username, type, resource):
+    resource = join(getUserSyncPath(context, username), resource)
+    waitForFileOrFolderToHaveSyncError(context, resource, type)
+
+
```

Comment:
```suggestion
@When(r'the user waits for (file|folder) "([^"]*)" to have sync error', regexp=True)
def step(context, type, resource):
    waitForFileOrFolderToHaveSyncError(context, resource, type)


@When(
    r'user "([^"]*)" waits for (file|folder) "([^"]*)" to have sync error', regexp=True
)
def step(context, username, type, resource):
    resource = join(getUserSyncPath(context, username), resource)
    waitForFileOrFolderToHaveSyncError(context, resource, type)


```
do we need both steps??

Reply:
Yes, client can have more than one user and in that case, we need to specify which user waits